src/target/TargetCollection.php

```diff
@@ -30,15 +32,17 @@ class TargetCollection implements TargetInterface
 
     public function __construct(array $targets)
     {
-        $this->targets = $targets;
         $ids = [];
         $types = [];
         $states = [];
         foreach ($targets as $target) {
-            if ($target) {
+            if ($target instanceof TargetInterface) {
                 $ids[] = $target->getId();
                 $types[] = $target->getType();
                 $states[] = $target->getState();
+                $this->targets[] = $target;
+            } else {
+                throw new UnknownEntityException('Target ' . $target . ' is incorrect');
```

Comment:
If the `$target` is not `string` or does not have `__toString()` method, this line will cause an exception itself.

How did you check this thing?

Reply:
My bad. I didn't check it. I will be more careful

Reply:
It's just a lack of typing. Once we clarify that `$targets` is an array of some specific type, PHPStorm will instantly let us know of suspicious type casts like this. That's why it's important to add types whenever you touch untyped code: this way we increase the whole code base stability.